cns/restserver/ipam.go

```diff
@@ -1098,11 +1098,11 @@ func (service *HTTPRestService) UpdateEndpointHelper(endpointID string, req cns.
 	if endpointInfo, ok := service.EndpointState[endpointID]; ok {
 		logger.Printf("[updateEndpoint] Found existing endpoint state for infra container %s", endpointID)
 		if req.HnsEndpointID != "" {
-			service.EndpointState[endpointID].HnsEndpointID = req.HnsEndpointID
+			service.EndpointState[endpointID].IfnameToIPMap[req.IFName].HnsEndpointID = req.HnsEndpointID
```

Comment:
Is there a scenario where `ifname` is not present in the `IfnameToIPMap` ?

Reply:
another condition will be added to check for empty interface.